charts/jobs/templates/job.yaml

```diff
@@ -29,36 +33,42 @@ spec:
       containers:
       - image: "{{ $spec.image.repository }}:{{ $spec.image.tag }}"
         imagePullPolicy: "{{ $spec.image.pullPolicy }}"
-        name: "{{ $name }}"
+        name: "{{ $spec.name }}"
         command: ["{{ $spec.shell | default "sh" }}", "-c"]
         args:
           - |
             {{- toString $spec.script | nindent 12 }}
         resources: {{- if $spec.resources}}{{ toYaml $spec.resources | nindent 10 }}{{ else }}{{ toYaml $v.resources | nindent 10 }}{{ end }}
+        {{- if or $spec.secret $spec.env }}
         envFrom:
         {{- if $spec.secret }}
         - secretRef:
-            name: {{ $name }}
+            name: {{ $spec.name }}
         {{- end }}
         {{- if $spec.env }}
         - configMapRef:
-            name: {{ $name }}
+            name: {{ $spec.name }}
         {{- end }}
+        {{- end }}
+        {{- if $spec.mounts }}
         volumeMounts:
+          {{- if $spec.fakeroot }}
           - name: fakeroot
             mountPath: /fakeroot.pem
             subPath: fakeroot.pem
-          {{- if $spec.mounts }}
-          {{- $spec.mounts | toYaml | nindent 10 }}
           {{- end }}
+          {{- $spec.mounts | toYaml | nindent 10 }}
```

Comment:
what happened to the condition? I would put it back to avoid issues.

And where is the logic to create the mounts? That whole part is missing.

Reply:
Do you mean the removal of line 51? It's moved to line 53 because otherwise in some cases there is an empty key `volumeMounts`. Jeho and I moved it when we were testing deployment.

Reply:
Actually, I understand the problem now, apologies.